bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/discovery/HaperoDiscoveryService.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hapero.internal.discovery;
+
+import java.util.Collections;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.hapero.internal.HaperoBindingConstants;
+import org.openhab.binding.hapero.internal.handler.HaperoBridgeHandler;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link HaperoDiscoveryService} is responsible for discovery of new devices.
+ *
+ * @author Daniel Walter - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = HaperoDiscoveryService.class, immediate = true, configurationPid = "discovery.hapero")
+public class HaperoDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+
+    private static final int DISCOVER_TIMEOUT_SECONDS = 5;
+    private @NonNullByDefault({}) HaperoBridgeHandler bridgeHandler;
+
+    /**
+     * Constructor
+     */
+    public HaperoDiscoveryService() {
+        super(Collections.singleton(new ThingTypeUID(HaperoBindingConstants.BINDING_ID, "-")), DISCOVER_TIMEOUT_SECONDS,
```

Comment:
You could use `Set.of(...)` or `List.of(...)` (depending upon superclass expectation) instead of `Collections.singleton()`

Reply:
This is how the skeleton script creates it, so i assume that is how it is supposed to be ?

Reply:
Skeleton could be updated, this reduces one dependency.